include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1084,45 +1086,40 @@ struct __early_exit_find_or
 {
     _Pred __pred;
 
-    template <typename _NDItemId, typename _IterSize, typename _WgSize, typename _LocalFoundState, typename _BrickTag,
-              typename... _Ranges>
+    template <typename _NDItemId, typename _SrcDataSize, typename _IterationDataSize, typename _LocalFoundState,
+              typename _BrickTag, typename... _Ranges>
     void
-    operator()(const _NDItemId __item_id, const _IterSize __n_iter, const _WgSize __wg_size,
-               _LocalFoundState& __found_local, _BrickTag __brick_tag, _Ranges&&... __rngs) const
+    operator()(const _NDItemId __item_id, const _SrcDataSize __source_data_size,
+               const _IterationDataSize __iteration_data_size, _LocalFoundState& __found_local, _BrickTag __brick_tag,
+               _Ranges&&... __rngs) const
     {
+        // Calculate the number of elements to be processed by each work-item.
+        using _Iteration =
+            decltype(oneapi::dpl::__internal::__dpl_ceiling_div(__source_data_size, __iteration_data_size));
+        const _Iteration __iters_per_work_item =
+            oneapi::dpl::__internal::__dpl_ceiling_div(__source_data_size, __iteration_data_size);
```

Comment:
Can we just use `auto` and `std::decltype(__iters_per_work_item)` to do this without repeating the expression?

Also, do we expect this type to be something other than `_SrcDataSize`?  It seems like that is what it will be.

Reply:
Fixed